### PR TITLE
remove memory leak

### PR DESCRIPTION
**Pull request by LaVillaStargato**

### CODE REVIEW

File: src/devMch.c [hunk 1]
```diff
@@ -308,16 +308,21 @@ MchDev rval = 0, d;
                          *       (_must_ pass d->name, not 'name'!!)
                          */
                         if ( registryAdd( registryId, d->name, d ) ) {
-                                rval = d; d = 0;
+                                /* success */
+                                rval = d;
+                                d = 0;
```
Comment (marciodo) [inline src/devMch.c:313]: Can't we just use one pointer here? What is the advantage of having 2 pointers for this operation?

File: src/drvMch.c [hunk 2]
```diff
@@ -2341,10 +2341,7 @@ int      inst;
 
 	mchStatMtx[inst] = epicsMutexMustCreate(); /* Used for global mchStat mask */
 
-	/* Allocate memory for MCH device support structure */
-	if ( ! (mch = calloc( 1, sizeof( *mch ) )) )
-		cantProceed("FATAL ERROR: No memory for MchDev structure\n");
-
```
Comment (marciodo) [inline src/drvMch.c:2347]: Were the tests ok after removing this?

File: src/devMch.c (hunk 1 above)
Comment (LaVillaStargato) [reply src/devMch.c:313]: There is no advantage.  I am just hesitant to change code that is working, so instead I added comments to make it clearer the intention of the code.

File: src/drvMch.c (hunk 2 above)
Comment (LaVillaStargato) [reply src/drvMch.c:2347]: Yes.

**Review by marciodo — APPROVED** [review]

**Pull request merged**